client/src/components/exercise/lab6/pages/Applicant/ExerciseStart.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable react/prop-types */
+import React, { useEffect } from "react";
+import { navigate } from "@reach/router";
+import { EXERCISE_IDLE, EXERCISE_PLAYING } from "../../../../../constants/lab6";
+import { bindActionCreators } from "redux";
+import { actions as exerciseActions } from "../../../../../reducers/lab6/ExerciseReducer";
+import { connect } from "react-redux";
+
+const mapStateToProps = (state) => ({
+  state: state,
+});
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    actions: bindActionCreators({ ...exerciseActions }, dispatch),
+  };
+};
+
+const ExerciseStart = (props) => {
+  const { actions } = props;
+
+  useEffect(() => {
+    actions.updateState(EXERCISE_IDLE);
+  }, [actions]);
+
+  const handleStart = () => {
+    actions.updateState(EXERCISE_PLAYING);
```

Comment:
Here, the state is updated to `EXERCISE_PLAYING` when the exercise first starts, as it should. In other components, a useEffect is used to update the state again to `EXERCISE_PLAYING` when the component is first loaded. I don't we are required to update the state after handleStart is invoked. I don't think the updating of the state in the other components is an issue, but I think we should remove it if it does nothing.

Reply:
I believe this is done because if they refresh the page the state is not saved? or if they decide to launch into the section from the url i want it to have the exercise playing state if that makes sense

Reply:
Ah, I don't have lab 7 setup that way. If they jump into a section, the state will be `EXERCISE_IDLE` since they didn't start from the initial page. Will update that.